### ISSUE TITLE
Retried requests duplicate  query parameters resulting in 400 response

### ISSUE THREAD

**Issue by geigerzaehler** [contentful/contentful.js#2246]
When a request is retried query parameters are duplicated which results in a 400 response.

This is a bit tricky to reproduce, but possible with Chrome Dev Tools’ [network request blocking](https://developer.chrome.com/docs/devtools/network-request-blocking) feature. I have a simple script that I load in the browser:

```
await client.getEntries({ contentType: "foo" })
```

In the dev tools I add “contentful“ to the list of blocked requests.
![image](https://github.com/contentful/contentful.js/assets/3919579/dae53f9b-1c21-4829-a6f9-e1b71e5f992f)

Then I reload the page. In the network tab of the dev tools I can see that requests to contentful are blocked. I quickly uncheck contentful in the request blocking tab to let it go through. But the server responds with a 400 and the message “The query you sent included duplicate URI query parameters...”.

![image](https://github.com/contentful/contentful.js/assets/3919579/fcd1ed7f-783b-47ad-b292-131584e06405)

From the network tab I can see that the request URL looks like this:
```
https://cdn.contentful.com/spaces/m2xydvefjw31/environments/master/entries?content_type=foo&content_type=foo&content_type=foo
```

It looks like every retry appends the query string again.

This issue started to occur in 10.11.1 and wasn’t present in 10.11.0.

**Comment by Techn1x** [comment]
We're experiencing the same issue from the same version and have put through a support request to contentful

**Comment by t-col** [comment]
Hi @geigerzaehler thanks for reporting. A stack trace of the `The query you sent included duplicate URI query parameters` error would be helpful in diagnosing where this problem is occurring. I have not been able to reproduce yet with the same setup you're describing.

**Comment by geigerzaehler** [comment]
Thanks for talking a look @t-col. Here’s the stack trace:
```
BadRequest
    at Ur (http://localhost:3000/node_modules/.vite/deps/contentful.js?v=9eaebbc3:6774:22)
    at a2 (http://localhost:3000/node_modules/.vite/deps/contentful.js?v=9eaebbc3:7217:17)
    at async u2 (http://localhost:3000/node_modules/.vite/deps/contentful.js?v=9eaebbc3:7224:27)
   [REDACTED]
```

It looks like the package does not provide any sourcemaps so this doesn’t look helpful unfortunately.

Can I make the steps to reproduce the issue clearer in any way? Which part can’t you reproduce?

**Comment by Techn1x** [comment]
If it helps, I was able to reproduce this with some simple steps:
- visit website
- go offline in Dev tools
- make contentful request (eg clicking a button that requests entries)
- in network tab, observe contentful requests that are retrying have duplicated query params

It's a supplemental step to then go back online in Dev tools and see that the next request when internet is available will return with 400 error from server because it has incorrect query params. The issue isn't so much to do with the error itself, but rather whatever is retrying the request is doing so incorrectly.

**Comment by t-col** [comment]
@geigerzaehler in the interest of thoroughness - are you passing in any custom configuration to the underlying axios instance from within the wrapping vite application? i.e. 
```
const client = createClient({
  space: '<your_space>',
  accessToken: '<your_token>',
  // ... other configuration
})
```

**Comment by geigerzaehler** [comment]
> are you passing in any custom configuration to the underlying axios instance from within the wrapping vite application?

No, I’m only passing space and access token to `createClient`.

**Comment by Techn1x** [comment]
repro video

https://github.com/contentful/contentful.js/assets/1049837/86a6a8c9-ed9a-4bf8-a003-276dd5c3e18d

here's the 400 response if you need it. It's BadRequest due to the duplicated query params.
![Screenshot 2024-06-12 at 7 17 05 pm](https://github.com/contentful/contentful.js/assets/1049837/4be6e8ec-407e-4cde-a998-d23bc01968b2)

**Comment by t-col** [comment]
fixed in [v10.11.11](https://github.com/contentful/contentful.js/compare/v10.11.10...v10.11.11)
